client/src/main/java/io/pravega/client/stream/impl/Controller.java

```diff
@@ -87,16 +88,40 @@
 
     /**
      * API to merge or split stream segments.
+     *
+     * @param stream Stream object.
+     * @param sealedSegments List of segments to be sealed.
+     * @param newKeyRanges Key ranges after scaling the stream.
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the scaling was started or not.
+     */
+    CompletableFuture<Boolean> startScale(final Stream stream, final List<Integer> sealedSegments,
+                                           final Map<Double, Double> newKeyRanges);
+
+    /**
+     * API to merge or split stream segments.
      * 
      * @param stream Stream object.
      * @param sealedSegments List of segments to be sealed.
      * @param newKeyRanges Key ranges after scaling the stream.
+     * @param timeout timeout and throw exception if scale does not complete withing specified timeout.
+     * @param executorService executor to be used for busy waiting.
```

Comment:
The original api used to block on rpc call to controller until scale completed.
scale stream retains the same semantics from the caller's perspective.
Note: we have added two new apis called startScale and checkScaleStatus if they want to do it themselves. 

And we still return a future to the caller from scaleStream. 
We essentially keep scheduling checkScale on the supplied executor till either scale completes or timeout happens. And upon either of those occurrences we complete the returned future.